### PR TITLE
Make bookmark toggle on NTP whether bookmark is empty or not

### PR DESCRIPTION
**Pull request by simonhong**
Fix https://github.com/brave/brave-browser/issues/1306

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
`yarn test brave_browser_tests --filter=BookmarkTabHelper*`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: patches/chrome-browser-ui-bookmarks-bookmark_tab_helper.cc.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc b/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc
+index f6ef61a0ac86ae55e8f9e80ece0758253af48520..888d656cf7080ebc32e90fc492e70a9b394d7026 100644
+--- a/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc
++++ b/chrome/browser/ui/bookmarks/bookmark_tab_helper.cc
+@@ -67,6 +67,7 @@ bool BookmarkTabHelper::ShouldShowBookmarkBar() const {
+       !prefs->GetBoolean(bookmarks::prefs::kShowBookmarkBar))
+     return false;
+ 
++  return false;
```
Comment (emerick) [inline patches/chrome-browser-ui-bookmarks-bookmark_tab_helper.cc.patch]: Was expecting to see something like `return !IsNTP(web_contents())` here. Is it safe to just return false here without further checks?

Comment (simonhong) [reply patches/chrome-browser-ui-bookmarks-bookmark_tab_helper.cc.patch]: I think this method is used for preventing bookmark bar visible or hidden opposite to the `kShowBookmarkBar` prefs value.

When `kShowBookmarkBar` is set to hidden, this method only returns true when active tab is NTP and bookmark bar isn't empty if all above conditions aren't met. So, we can hide bookmark bar on NTP by returning false in this case.

When `kShowBookmarkBar` is set to visible, this method isn't called except fullscreen condition.
According to the comment of `Browser::ShouldHideUIForFullscreen()`, browser controls UI can be visible on MacOS fullscreen mode by sliding down. When `Browser::ShouldHideUIForFullscreen()` returns true, this method will be called and we should also return false to hide bookmark bar.

So, I think returning false in here is safe.

Comment (emerick) [reply patches/chrome-browser-ui-bookmarks-bookmark_tab_helper.cc.patch]: OK, got it.

Comment (simonhong) [reply patches/chrome-browser-ui-bookmarks-bookmark_tab_helper.cc.patch]: @emerick Thanks for quick review!

**Pull request merged**